### PR TITLE
sanitycheck: Remove duplicate headers in CSV file and add named exceptions

### PR DESCRIPTION
**Pull request by nashif**
Dont merge test files with multiple header rows. 
Cleanup exceptions and logging.

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -2646,7 +2646,8 @@ class TestSuite:
                 cr = csv.DictReader(fp)
                 instance_list = []
                 for row in cr:
-                    if row["status"] in filter_status:
+                    # FIXME: ignore header row.
```
Comment (galak) [inline scripts/sanitycheck]: Can we explain the `row['arch'] == 'arch'` in the comment in a little more detail.

Something like:

```
# We might have header rows intermixed, detect them by the the 'arch' field being set to 'arch'
```

**Review by galak — APPROVED** [review]

File: scripts/ci/run_ci.sh [hunk 2]
```diff
@@ -251,7 +251,10 @@ if [ -n "$main_ci" ]; then
 
 	# Save list of tests to be run
 	${sanitycheck} ${sanitycheck_options} --save-tests test_file_3.txt || exit 1
-	cat test_file_1.txt test_file_2.txt test_file_3.txt > test_file.txt
+
+	tail -n +2 test_file_2.txt > test_file_2_in.txt
```
Comment (galak) [inline scripts/ci/run_ci.sh]: can you add a comment here about skipping the header.

File: scripts/sanitycheck [hunk 3]
```diff
@@ -2644,10 +2644,13 @@ class TestSuite:
             total_tests = 0
             with open(file, "r") as fp:
                 cr = csv.DictReader(fp)
+                # Skip header
+                next(cr)
```
Comment (galak) [inline scripts/sanitycheck]: Is this really needed, I thought the DictReader will pull off the header?

Comment (nashif) [reply scripts/sanitycheck]: it is not, with the above change

**Review by galak — APPROVED** [review]

**Pull request merged**